controllers/gcpmachine_controller_unit_test.go

```diff
@@ -81,23 +80,21 @@ func TestGCPMachineReconciler_GCPClusterToGCPMachines(t *testing.T) {
 		newMachine(clusterName, "my-machine-2"),
 	}
 
-	client := fake.NewFakeClientWithScheme(scheme, initObjects...)
+	client := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(initObjects...).Build()
```

Comment:
Does `WithObjects()` work here instead of `WithRuntimeObjects()`? I'm not sure if there is a behavioral change that we would need to worry about between the two...

Reply:
I think should be with `WithObjects()` will update

Reply:
looking at the other provide (azure) they use this  `WithRuntimeObjects`